pkg/gofr/logging/logger_test.go

```diff
@@ -104,29 +107,57 @@ func TestLogger_LevelWarn(t *testing.T) {
 	infoLog := testutil.StdoutOutputForFunc(printLog)
 	errLog := testutil.StderrOutputForFunc(printLog)
 
-	if strings.ContainsAny(infoLog, "NOTICE|INFO|DEBUG") && !strings.Contains(errLog, "ERROR") {
-		// Warn Log Level will not contain  DEBUG,INFO, NOTICE logs
-		t.Errorf("TestLogger_LevelDebug Failed!")
+	levels := []Level{DEBUG, INFO, NOTICE}
+
+	for i, l := range levels {
+		assert.Equal(t, false, strings.Contains(infoLog, l.String()), "TEST[%d], Failed.\n%s", i,
+			fmt.Sprintf("unexpected %s log", l))
 	}
 
 	assertMessageInJSONLog(t, errLog, "Test Error Log")
 }
 
 func TestLogger_LevelFatal(t *testing.T) {
-	printLog := func() {
+	// running the failing part only when a specific env variable is set
```

Comment:
This comment is explaining the what, not the why

maybe  something like this would be better


```suggestion
	// running the failing part only when a specific env variable is set
  // we are using this because it's one of the only way we found to test the Fatal
```